apps/consoledot-rhosak/src/routes/data-plane/routes/TopicCreateRoute.tsx

```diff
@@ -0,0 +1,110 @@
+import { useDataPlaneGate } from "../useDataPlaneGate";
+import type { DataPlaneNavigationProps } from "../routesConsts";
+import {
+  developerDefaults,
+  standardDefaults,
+} from "consoledot-api/src/transformers/topicTransformer";
+
+export const TopicCreateRoute: VoidFunctionComponent<
+  DataPlaneNavigationProps
+> = ({ instancesHref, instanceTopicsHref }) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const history = useHistory();
+  const { instance } = useDataPlaneGate();
+  const createTopic = useCreateTopicMutation();
+  const dispatch = useDispatch();
+  const availabiltyZone = instance.az;
+  const { data: topics } = useTopics({
+    id: instance.id,
+    adminUrl: instance.adminUrl,
+    plan: instance.plan,
+  });
+
+  const checkTopicName = (topicName: string) => {
+    const selectedTopic = topics?.topics.filter(
+      (topic) => topic.name === topicName
+    );
+    if (selectedTopic && selectedTopic?.length > 0) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const onCloseCreateTopic = useCallback(() => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+    history.push(instanceTopicsHref(instance.id));
+  }, [history, instance.id, instanceTopicsHref]);
+
+  const onSave = useCallback(
+    (topicData: Topic) => {
+      void createTopic.mutateAsync({
+        instanceId: instance.id,
+        adminUrl: instance?.adminUrl || "",
+        topic: topicData,
+
+        onSuccess: () => {
+          //eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+          history.push(instanceTopicsHref(instance.id));
+        },
+        onError: (_, message) => {
+          dispatch(
+            addNotification({
+              variant: "danger",
+              title: message,
+              dismissable: true,
+              id: "save-error",
+            })
+          );
+        },
+      });
+    },
+    [
+      createTopic,
+      dispatch,
+      history,
+      instance?.adminUrl,
+      instance.id,
+      instanceTopicsHref,
+    ]
+  );
+
+  const initialTopicValues: Topic =
+    instance.plan === "developer"
+      ? {
+          name: "",
+          partitions: [{ partition: 1, id: 1 }],
+          ...developerDefaults,
+        }
+      : {
+          name: "",
+          partitions: [{ partition: 1, id: 1 }],
+          ...standardDefaults,
+        };
+
+  return (
+    <>
+      <CreateTopic
+        kafkaName={instance.name}
+        onKafkaPageLink={instanceTopicsHref(instance.id)}
+        onKafkaInstanceLink={instancesHref}
+        onSave={onSave}
+        initialTopicValues={initialTopicValues}
+        onCloseCreateTopic={onCloseCreateTopic}
+        checkTopicName={checkTopicName}
+        availablePartitionLimit={
+          instance.maxPartitions || instance.plan == "developer" ? 100 : 1000
```

Comment:
Why is this needed? The Kafka object transformer should provide the right value without having to place this logic here. If the problem is that the value can be undefined, add a defensive check before the return:

```ts
if (instance.maxPartitions === undefined) {
    throw new Error(`CreateTopicRoute, unexpected maxPartition undefined for instance ${instance.name}`);
}
```

Or better, challenge the fact that we have those undefined in the first place. This is probably a legacy from the old codebase, but in this new one the Kafka object is prepared with all the info all the time (but for a few bits like the URLs that the API provides at a later moment)